### PR TITLE
CVE-2011-3079

### PR DESCRIPTION
**Pull request by thatapplefreak**

### CODE REVIEW

File: cves/CVE-2011-3079.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 CVE: CVE-2011-3079
-CWE: 
+CWE:
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: You need to find a CWE that corresponds to this CVE

File: cves/CVE-2011-3079.yml [hunk 2]
```diff
@@ -15,10 +15,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Inter-process Communication implementation does not correctly validate
+  messages allowing for attack vectors. It is possible to create a fake
+  listener process that does not have the propper permissions to connect to
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: Spelling

File: cves/CVE-2011-3079.yml [hunk 3]
```diff
@@ -27,22 +31,27 @@ reviews:
...
 fixes:
+- :commit: 48fae61b8a6c9b741f001d478c595b6c7c0af4d9
+  :note: 'sandboxed processes cannot open each other'
 - :commit: 0590878f77dd9d3791b936287db516a99d14efb0
-  :note: ''
-vccs: []
+  :note: 'requires message to be validated by IsHelloMessage'
+vccs:
+- :commit: 4734d0becafa5b77d708020eed24d97148ea208d
+  :note: 'the vulerability was created by creating a random process id rather
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: Spelling

File: cves/CVE-2011-3079.yml [hunk 4]
```diff
@@ -55,9 +64,11 @@ unit_tested:
 
     For the "fix" answer below, check if the fix for the vulnerability involves
     adding or improving an automated test to ensure this doesn't happen again.
-  answer: 
-  code: 
-  fix: 
+  answer: |
+    Tests were added to check for the vulenrability in commit
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: Spelling

File: cves/CVE-2011-3079.yml [hunk 5]
```diff
@@ -73,20 +84,24 @@ discovered:
...
-  answer: 
-  date: 
-  automated: 
-  google: 
-  contest: 
+  answer: |
+    Comments regarding details of how this vulenrability was found have been
+    deleted.
+  date: 2012-03-10
+  automated:
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: Mark these as false or nil if they are not found through that avenue.

File: cves/CVE-2011-3079.yml [hunk 6]
```diff
@@ -73,20 +84,24 @@ discovered:
...
   question: |
     What subsystems was the mistake in?
 
     Look at the path of the source code files code that were fixed to get
     directory names. Look at comments in the code. Look at the bug reports how
     the bug report was tagged.
-  answer: 
-  name: 
+  answer: |
+    Vulnerability seems to be in the browser's redering host as the affected
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: Spelling

File: cves/CVE-2011-3079.yml [hunk 7]
```diff
@@ -129,38 +144,41 @@ lessons:
     If you think of another lesson we covered in class that applies here, feel
     free to give it a small name and add one in the same format as these.
   defense_in_depth:
-    applies: 
-    note: 
+    applies: true
+    note: |
+      The fix involved having a secert hello message.
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: This needs more information and correct spelling. How is this defense in depth? What are the different layers of security?

File: cves/CVE-2011-3079.yml [hunk 8]
```diff
@@ -173,4 +191,10 @@ mistakes:
...
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer: 
+  answer: |
+    The mistake was trusting the pipe of input given to the process. By just 
+    assigning a random id to the process and adding it the system was trusting
+    it by default.
+    
+    It was introduced when child process hosts were refactored through an
+    interface in content/public rather than inheritance.
```
Comment (laura-king) [inline cves/CVE-2011-3079.yml]: This needs more information. How did the fix work? How can this be prevented in the future?

**Comment by dxf7606** [comment]
As Laura has pointed out, there's a lot of spelling mistakes. I think the content is fine otherwise

File: cves/CVE-2011-3079.yml (hunk 2 above)
Comment (JurelCastillo) [inline cves/CVE-2011-3079.yml]: Proper*

File: cves/CVE-2011-3079.yml [hunk 9]
```diff
@@ -15,10 +15,14 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description: 
+description: |
+  Inter-process Communication implementation does not correctly validate
+  messages allowing for attack vectors. It is possible to create a fake
+  listener process that does not have the propper permissions to connect to
+  a pipe started by the browser's process and cause a privilege escalation.
```
Comment (JurelCastillo) [inline cves/CVE-2011-3079.yml]: privledge*

File: cves/CVE-2011-3079.yml [hunk 10]
```diff
@@ -15,10 +15,14 @@ description_instructions: |
...
 bounty:
```
Comment (JurelCastillo) [inline cves/CVE-2011-3079.yml]: no bounty? found by google?

File: cves/CVE-2011-3079.yml (hunk 3 above)
Comment (JurelCastillo) [inline cves/CVE-2011-3079.yml]: vulnerability* I'd*

File: cves/CVE-2011-3079.yml (hunk 4 above)
Comment (JurelCastillo) [inline cves/CVE-2011-3079.yml]: vulnerability*

File: cves/CVE-2011-3079.yml (hunk 6 above)
Comment (JurelCastillo) [inline cves/CVE-2011-3079.yml]: rendering*
You spelled vulnerability correctly though!

File: cves/CVE-2011-3079.yml (hunk 7 above)
Comment (JurelCastillo) [inline cves/CVE-2011-3079.yml]: secret*
I would agree with it's a defense in depth less, but could you give a better note?

**Comment by sei-dleung** [comment]
@andymeneely I believe https://github.com/chromium/chromium/commit/215f6fd17639a28b070af3bf06cd70402ec84cab is the actual VCC for this vulnerability. This is the commit which adds WillDispatchInputMessage() which always returns true. The fix subsequently adds a validation check before returning the boolean. The current commit set as the VCC doesn't touch ipc/ipc_channel_win.cc which is where the distrust input vulnerability exists.

**Pull request merged**